guides/complexity-analysis.md

```diff
@@ -1,7 +1,19 @@
-# Complexity Analysis
+# Safety Limits
 
 A misbehaving client might send a very complex GraphQL query that would require
-considerable resources to handle. In order to protect against this scenario, the
+considerable resources to handle. There are two variations of this problem:
+
+- Complex queries that overwhelm resolution resources.
+- Extremely long queries that could take considerable resources to parse. 
+(For example, an attacker could craft a long query including thousands of 
+undefined fields or directives.)
+
+Either of these could be a vector for a denial-of-service attack in any GraphQL 
+service. Absinthe includes mechanisms to protect services each of these.
+
+## Complexity Analysis
+
+In order to protect against queries that could overwhel scenario, the
```

Comment:
```suggestion
To protect against queries that could overwhelm available resources, the
```

Reply:
Whoops, not sure how that happened! 😵‍💫 Thank you!